### PR TITLE
[ray_client] add client microbenchmarks

### LINKED ISSUE #11980: Add client metrics / Add to `ray microbenchmark` suite [3]
[sync with @simon-mo ]?

### PR DESCRIPTION
**Pull request by barakmich**
## Related issue number

Depends on #12985
Closes #11980 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/ray_client_helpers.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from contextlib import contextmanager
```
Comment (ericl) [inline python/ray/_private/ray_client_helpers.py:1]: Consider keeping this file in `ray/experimental/client`

**Review by ericl — COMMENTED** [review]
Can you post the new output of `ray microbenchmark` (laptop run is fine)?

File: python/ray/_private/ray_client_helpers.py (hunk 1 above)
Comment (barakmich) [reply python/ray/_private/ray_client_helpers.py:1]: That's a good idea

**Comment by barakmich** [comment]
Rebased. Here's the test output:
```
(ray) (18:47|333)[barak@theseus ~/src/ray/python]
$ ray microbenchmark
Tip: set TESTS_TO_RUN='pattern' to run a subset of benchmarks
2020-12-21 18:47:36,196 INFO services.py:1169 -- View the Ray dashboard at http://127.0.0.1:8265
single client get calls per second 65521.93 +- 5711.67
single client put calls per second 37810.0 +- 1123.98
multi client put calls per second 153221.39 +- 1493.19
2020-12-21 18:48:06,728 INFO services.py:1169 -- View the Ray dashboard at http://127.0.0.1:8265
single client get calls (Plasma Store) per second 12660.34 +- 213.58
single client put calls (Plasma Store) per second 5132.93 +- 23.37
multi client put calls (Plasma Store) per second 7639.78 +- 43.95
single client put gigabytes per second 16.26 +- 9.15
multi client put gigabytes per second 20.94 +- 0.16
single client tasks sync per second 1483.83 +- 22.12
single client tasks async per second 15862.7 +- 444.34
multi client tasks async per second 23957.43 +- 409.69
1:1 actor calls sync per second 2621.91 +- 51.88
1:1 actor calls async per second 9112.33 +- 483.0
1:1 actor calls concurrent per second 7229.69 +- 148.99
1:n actor calls async per second 17260.06 +- 312.05
n:n actor calls async per second 31760.54 +- 617.17
n:n actor calls with arg async per second 9107.89 +- 31.47
1:1 async-actor calls sync per second 1842.6 +- 64.04
1:1 async-actor calls async per second 5347.39 +- 153.14
1:1 async-actor calls with args async per second 3626.96 +- 210.17
1:n async-actor calls async per second 13585.07 +- 316.52
n:n async-actor calls async per second 20920.13 +- 427.89
client: get calls per second 2047.28 +- 16.5
client: put calls per second 1029.47 +- 22.16
client: remote put calls per second 60132.68 +- 381.32
client: 1:1 actor calls sync per second 635.62 +- 6.61
client: 1:1 actor calls async per second 663.8 +- 9.61
client: 1:1 actor calls concurrent per second 666.98 +- 7.88
```

**Review by ericl — APPROVED** [review]
> client: remote put calls per second 60132.68 +- 381.32

Interesting, this is faster than the baseline. Is it since we don't wait for an ack?

Edit: ah nevermind, it's in a remote fn, makes sense

**Pull request merged**